setup.py

```diff
@@ -107,9 +108,17 @@ def get_extensions():
     extensions_cuda_dir = os.path.join(extensions_dir, "cuda")
     cuda_sources = list(glob.glob(os.path.join(extensions_cuda_dir, "**/*.cu"), recursive=True))
 
-    if use_cuda:
+    extensions_hip_dir = os.path.join(extensions_dir, "cuda", "tensor_core_tiled_layout")
+    hip_sources = list(glob.glob(os.path.join(extensions_hip_dir, "*.cu"), recursive=True))
```

Comment:
hmm kind of a weird way to structure the directories, ideally we should have something like `csrc/hip`, `csrc/cuda` - I'd be OK with some code duplication fwiw

Reply:
Hi @msaroufim Thanks for taking a look and providing feedback. Currently CUDAExtensions internally uses torch.utils.hipify which automatically converts CUDA code to HIP code. So when we install CUDAExtensions on ROCm platform it automatically creates a csrc/hip folder with only tensor core hipified in this case.  

Once, we have enabled all the custom ops, we plan to remove the hip_sources variable and directly use cuda_sources. 

Example implementation: https://github.com/pytorch/vision/blob/main/setup.py

cc: @petrex

Reply:
> hmm kind of a weird way to structure the directories, ideally we should have something like `csrc/hip`, `csrc/cuda` - I'd be OK with some code duplication fwiw

thanks, once we ran `python setup.py install`, we shall see hip kernels generated under csrc/hip.